### PR TITLE
fix: block does not merge with player

### PR DESCRIPTION
**Pull request by nicholaschuayunzhi**
Place an inner rigidbody for all blocks so that even when player clips slightly with ground, the inner block will still interact with the player.

Will probably have to use raycasts in the future.

### CODE REVIEW

File: Assets/Scripts/BlockFaceBehaviour.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@ public class BlockFaceBehaviour : MonoBehaviour
 {
 
     public BlockBehaviour Block;
+    public LayerMask IgnoreCollision;
```
Comment (Rinder5) [inline Assets/Scripts/BlockFaceBehaviour.cs]: what is this variable used for?

File: Assets/Scripts/BlockBehaviour.cs [hunk 2]
```diff
@@ -15,6 +15,8 @@ public class BlockBehaviour : MonoBehaviour
     public BlockFaceBehaviour Left;
     public BlockFaceBehaviour Right;
 
+    public LayerMask IgnoreCollision;
```
Comment (Rinder5) [inline Assets/Scripts/BlockBehaviour.cs:18]: same here

File: Assets/Scripts/BlockFaceBehaviour.cs (hunk 1 above)
Comment (nicholaschuayunzhi) [reply Assets/Scripts/BlockFaceBehaviour.cs]: ahh this is a mistake.

File: Assets/Scripts/BlockBehaviour.cs (hunk 2 above)
Comment (nicholaschuayunzhi) [reply Assets/Scripts/BlockBehaviour.cs:18]: For this, a raycast that fires out from a block will ignore anything selected in the layer mask. In the inspector, i have selected for it to ignore `Player`. The block will push players.

Comment (Rinder5) [reply Assets/Scripts/BlockBehaviour.cs:18]: ok. so it needs to be set for each block which we place in a level? will we ever need it to ignore more than 1 layer?

Comment (nicholaschuayunzhi) [reply Assets/Scripts/BlockBehaviour.cs:18]: We can set more than 1 layer in the layer mask. At the same time, the ignore can be set at the prefab level, to affect all blocks.

Comment (Rinder5) [reply Assets/Scripts/BlockBehaviour.cs:18]: kk, sounds good 👍

**Pull request merged**